src/main/java/com/patterns/method/ProxyMacOS.java

```diff
@@ -0,0 +1,18 @@
+package com.patterns.method;
+
+public class ProxyMacOS implements Os {
```

Comment:
Also, read about another dynamic way to implement a proxy. 
https://www.baeldung.com/java-dynamic-proxies

Reply:
Thanks for the link. I will definitely read it